social-messenger-ts/src/hooks/useMobileView.ts

```diff
@@ -0,0 +1,22 @@
+import { useCallback, useEffect, useState } from 'react';
+
+/**
+ * Internal, handles the mobile/responsive view adjustments.
+ */
+export const useMobileView = () => {
+  const [isMobileNavVisible, setMobileNav] = useState(false);
+  useEffect(() => {
+    const mobileChannelList = document.querySelector('#mobile-channel-list');
+    if (isMobileNavVisible && mobileChannelList) {
+      mobileChannelList.classList.add('show');
+      document.body.style.overflow = 'hidden';
+    } else if (!isMobileNavVisible && mobileChannelList) {
+      mobileChannelList.classList.remove('show');
+      document.body.style.overflow = 'auto';
+    }
```

Comment:
This feels weird though I'm not sure if it's worth refactoring - I think CSS variables would be a better solution for this.